src/util.rs

```diff
@@ -0,0 +1,36 @@
+use crate::token::TokenType;
+
+pub fn is_digit(s: &str) -> bool {
```

Comment:
use unicode categories?

tho you'd need to do some weird shit to get the decimal value of the digit